pkg/webhook/plugins/requirenodewithfuse/require_node_with_fuse.go

```diff
@@ -0,0 +1,108 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requirenodewithfuse
+
+import (
+	"fmt"
+
+	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+/*
+   This plugin is for pods with a  dataset.
+   They should require nods with fuse.
+*/
+
+const NAME = "RequireNodeWithFuse"
+
+type RequireNodeWithFuse struct {
+	client client.Client
+	name   string
+}
+
+func NewPlugin(c client.Client) *RequireNodeWithFuse {
+	return &RequireNodeWithFuse{
+		client: c,
+		name:   NAME,
+	}
+}
+
+func (p *RequireNodeWithFuse) GetName() string {
+	return p.name
+}
+
+// Mutate mutates the pod based on runtimeInfo, this action shouldn't stop other handler
+func (p *RequireNodeWithFuse) Mutate(pod *corev1.Pod, runtimeInfos []base.RuntimeInfoInterface) (shouldStop bool, err error) {
+	// if the pod has no mounted datasets, should exit and call other plugins
+	if len(runtimeInfos) == 0 {
+		return
+	}
+
+	requiredSchedulingTerms := []corev1.NodeSelectorTerm{}
+
+	for _, runtime := range runtimeInfos {
+		term, err := getRequiredSchedulingTerm(runtime)
+		if err != nil {
+			return true, fmt.Errorf("Should stop mutating pod %s in namespace %s due to %v",
+				pod.Name,
+				pod.Namespace,
+				err)
+		}
+
+		requiredSchedulingTerms = append(requiredSchedulingTerms, term)
+	}
+
+	utils.InjectNodeSelectorTerms(requiredSchedulingTerms, pod)
+
+	return
+}
+
+func getRequiredSchedulingTerm(runtimeInfo base.RuntimeInfoInterface) (requiredSchedulingTerms corev1.NodeSelectorTerm, err error) {
+	requiredSchedulingTerms = corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{},
+	}
+
+	if runtimeInfo == nil {
+		err = fmt.Errorf("RuntimeInfo is nil")
+		return
+	}
+
+	isGlobalMode, selectors := runtimeInfo.GetFuseDeployMode()
```

Comment:
why when fuse is not global, no selectors for fuse pod.

Reply:
When fuse is not global, the selector is the same with fluid worker.